src/service/viewer-impl.js

```diff
@@ -998,92 +1017,87 @@ export class Viewer {
    * @param {string} eventType
    * @param {*} data
    * @param {boolean} awaitResponse
-   * @return {!Promise<*>|undefined}
+   * @return {!Promise<*>|undefined} the response promise if awaitResponse is
+   *     true, otherwise undefined
    */
   sendMessage(eventType, data, awaitResponse) {
     if (!this.messagingReadyPromise_) {
       return Promise.reject(getChannelError());
     }
     return this.messagingReadyPromise_.then(() => {
-      return this.sendMessageUnreliable_(eventType, data, awaitResponse);
+      return this.messageDeliverer_(eventType, data, awaitResponse);
     });
   }
 
   /**
-   * Broadcasts a message to all other AMP documents under the same viewer. It
-   * will attempt to deliver messages when the messaging channel has been
-   * established, but it will not fail if the channel is timed out.
+   * Sends the message to the viewer. This method queues up the message if the
+   * communication channel isn't established yet. The message will cancel an
+   * unsent message of the same type if seen in the queue.
    *
-   * @param {!JSONType} message
-   */
-  broadcast(message) {
-    this.maybeSendMessage_('broadcast', message);
-  }
-
-  /**
-   * Registers receiver for the broadcast events.
-   * @param {function(!JSONType)} handler
-   * @return {!UnlistenDef}
-   */
-  onBroadcast(handler) {
-    return this.broadcastObservable_.add(handler);
-  }
-
-  /**
-   * This message queues up the message to be sent when communication channel
-   * is established. If the communication channel is not established at
-   * this time, this method responds immediately with a Promise that yields
-   * `undefined` value.
    * @param {string} eventType
    * @param {*} data
    * @param {boolean} awaitResponse
-   * @return {?Promise<*>|undefined}
-   * @private
+   * @return {?Promise<*>|undefined} the response promise if awaitResponse is
+   *     true, otherwise undefined
    */
-  sendMessageUnreliable_(eventType, data, awaitResponse) {
+  sendMessageCancelUnsent(eventType, data, awaitResponse) {
```

Comment:
it's just how the original `messageDeliverer` function is defined, which can return `undefined`. It doesn't really matter.

I see one little advantage of returning `undefined` is that it will throw immediately if you made mistake of awaitResponse=false when it should be true.